### PR TITLE
Migration: typescript/okta commands.js

### PR DESCRIPTION
**Pull request by hereje**
#### Summary

Convert file ` okta_commands.js` to typescript

- convert file to ts
- reorganize code
- add docs
- add types

#### Ticket Link

Fixes https://github.com/mattermost/mattermost/issues/21612

#### Release Note

```release-note
NONE
```

### CODE REVIEW

File: e2e-tests/cypress/tests/support/okta_commands.ts [hunk 1]
```diff
@@ -1,12 +1,47 @@
...
+import {ChainableT} from 'tests/types';
 import * as TIMEOUTS from '../fixtures/timeouts';
 
 const token = 'SSWS ' + Cypress.env('oktaMMAppToken');
 
-function buildProfile(user) {
-    const profile = {
+type UserId = string | null;
+
+interface Profile {
```
Comment (larkox) [inline e2e-tests/cypress/tests/support/okta_commands.ts:11]: Profile and User seems to be the same data structure, right? They only change on the whether they have `password` or not. Any strong reason to keep them separated?

File: e2e-tests/cypress/tests/support/okta_commands.ts [hunk 2]
```diff
@@ -35,29 +75,44 @@ Cypress.Commands.add('oktaCreateUser', (user = {}) => {
                 },
             },
         },
-    }).then((response) => {
+    }).then((response: any) => {
```
Comment (larkox) [inline e2e-tests/cypress/tests/support/okta_commands.ts:78]: Any chance we can get rid of this any? Can we type the responses?

Looking on other files, seems like the response for a `task` is by default unknown. Two ways to solve this is either assigning the type (that is not much type safer than casting) (e.g. `cy.task<SomeType>(...).then(...)`) or using guards.

0/5, but would be great to remove these any.

**Review by larkox — CHANGES_REQUESTED** [review]
I would love to remove the `any` from this file, but let me know if it is not possible for any reason.

File: e2e-tests/cypress/tests/support/okta_commands.ts (hunk 2 above)
Comment (hereje) [reply e2e-tests/cypress/tests/support/okta_commands.ts:78]: Sure!

File: e2e-tests/cypress/tests/support/okta_commands.ts (hunk 1 above)
Comment (hereje) [reply e2e-tests/cypress/tests/support/okta_commands.ts:11]: Even though both interfaces are similar, they differ in the property names in camel-case

`firstName` vs `firstname`

**Review by larkox — APPROVED** [review]

**Review by saturninoabril — APPROVED** [review]
Thank you @hereje for your contribution!

**Comment by hanzei** [comment]
@saturninoabril Are we good to merge the PR?

**Comment by saturninoabril** [comment]
Thank you @hereje!

**Pull request merged**